PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java

```diff
@@ -0,0 +1,154 @@
+
+        final BiConsumer<IAzureBaseResource<?, ?>, AnActionEvent> showFunctionPropertyViewHandler = (c, e) -> AzureTaskManager.getInstance()
+                .runLater(() -> new OpenAppServicePropertyViewAction().openFunctionAppPropertyView((FunctionApp) c, e.getProject()));
+        am.registerHandler(ResourceCommonActionsContributor.SHOW_PROPERTIES, (r, e) -> r instanceof FunctionApp, showFunctionPropertyViewHandler);
+
+        final BiConsumer<IAzureBaseResource<?, ?>, AnActionEvent> showWebAppPropertyViewHandler = (c, e) -> AzureTaskManager.getInstance()
+                .runLater(() -> new OpenAppServicePropertyViewAction().openWebAppPropertyView((WebApp) c, e.getProject()));
+        am.registerHandler(ResourceCommonActionsContributor.SHOW_PROPERTIES, (r, e) -> r instanceof WebApp, showWebAppPropertyViewHandler);
+
+        final BiConsumer<IAzureBaseResource<?, ?>, AnActionEvent> showWebAppSlotPropertyViewHandler = (c, e) -> AzureTaskManager.getInstance()
+                .runLater(() -> new OpenAppServicePropertyViewAction().openDeploymentSlotPropertyView((WebAppDeploymentSlot) c, e.getProject()));
+        am.registerHandler(ResourceCommonActionsContributor.SHOW_PROPERTIES, (r, e) -> r instanceof WebAppDeploymentSlot, showWebAppSlotPropertyViewHandler);
+
+        final BiPredicate<FunctionEntity, AnActionEvent> triggerPredicate = (r, e) -> r instanceof FunctionEntity;
+        final BiConsumer<FunctionEntity, AnActionEvent> triggerFunctionHandler = (entity, e) -> {
+            final String functionId = Optional.ofNullable(entity.getFunctionAppId())
+                    .orElseGet(() -> ResourceId.fromString(entity.getTriggerId()).parent().id());
+            final FunctionApp functionApp = Azure.az(AzureFunction.class).get(functionId);
+            final String triggerType = Optional.ofNullable(entity.getTrigger())
+                    .map(functionTrigger -> functionTrigger.getProperty("type")).orElse(null);
+            final Object request;
+            if (StringUtils.equalsIgnoreCase(triggerType, "timertrigger")) {
+                request = new Object();
+            } else {
+                final String input = AzureTaskManager.getInstance().runAndWaitAsObservable(new AzureTask<>(() -> Messages.showInputDialog(e.getProject(), "Please set the input value: ",
+                        String.format("Trigger function %s", entity.getName()), null))).toBlocking().single();
+                if (input == null) {
+                    return;
+                }
+                request = new TriggerRequest(input);
+            }
+            functionApp.triggerFunction(entity.getName(), request);
+        };
+        am.registerHandler(FunctionAppActionsContributor.TRIGGER_FUNCTION, triggerPredicate, triggerFunctionHandler);
+    }
+
+    @Override
+    public int getOrder() {
+        return INITIALIZE_ORDER;
+    }
+
+    @RequiredArgsConstructor
+    static class TriggerRequest {
```

Comment:
consider move TriggerRequest to triggerFunction

Reply:
Will refactor it after refactor for app service library